tests/xml_file_loader_tests.cc

```diff
@@ -71,6 +75,20 @@ std::string XMLFileLoaderTests::ControlSequenceWithDecay() {
           "</simulation>";
 }
 
+std::string XMLFileLoaderTests::ControlSequenceWithSolver() {
+  return  "<simulation>"
+          " <control>"
+          "  <duration>1200</duration>"
+          "  <startmonth>1</startmonth>"
+          "  <startyear>2000</startyear>"
+          "  <simstart>0</simstart>"
+          "  <solver>greedy</solver>"
```

Comment:
Perhaps I've misunderstood the schema as defined, but shouldn't this test use a nested block like 
```
<solver>
   <greedy>
       <preconditioner/>
   </greedy>
   <exclusive_orders_only/>
</solver>
```
?